### PR TITLE
Nfs alias resource

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
nfs alias resource and datasource

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 80% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Unit tests
- 
nfs alias UT

- [X] Acceptance tests
- 
nfs Alias AT

### CODE REVIEW

**Comment by Krunal-Thakkar** [comment]
UT For NFS Alias Resource And Datasource

image

AT

image


Code Coverage

image

**Comment by AnikaAgiwal2711** [comment]
Code coverage 
image

File: powerscale/provider/nfs_alias_resource.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+		return
+	}
+
+	diags := helper.CreateNfsAlias(ctx, r.client, plan, &state)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
```
Comment (shenda1) [inline powerscale/provider/nfs_alias_resource.go:131]: If possible, log the completion operation as well

File: powerscale/provider/nfs_alias_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+func TestAccNfsAliasResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read testing
+			{
+				Config: ProviderConfig + NfsAliasResourceConfig,
+				Check:  resource.ComposeAggregateTestCheckFunc(),
```
Comment (shenda1) [inline powerscale/provider/nfs_alias_resource_test.go:37]: Add the assertion statements

File: powerscale/provider/nfs_alias_datasource_test.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+				ExpectError: regexp.MustCompile("mock error"),
+			},
+		},
+	})
+}
+
+var NfsAliasDatasourceConfig = `
+data "powerscale_nfs_alias" "export_datasource_test" {
+	filter {
+		ids = ["alias1"]
```
Comment (shenda1) [inline powerscale/provider/nfs_alias_datasource_test.go:99]: Is alias1 is the pre-req ? If yes, can we create on fly?

Comment (Krunal-Thakkar) [reply powerscale/provider/nfs_alias_datasource_test.go]: removed, not using that

**Review by shenda1 — APPROVED** [review]

**Review by rounak-adhikary — APPROVED** [review]

**Comment by Krunal-Thakkar** [comment]
run e2e test NfsAliasResource

**Pull request merged**